tracing-subscriber/src/layer.rs

```diff
@@ -808,6 +808,92 @@ where
     }
 }
 
+impl<L, S> Layer<S> for Option<L>
+where
+    L: Layer<S>,
+    S: Subscriber,
+{
+    #[inline]
+    fn new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.new_span(attrs, id, ctx)
+        }
+    }
+
+    #[inline]
+    fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
+        match self {
+            Some(ref inner) => inner.register_callsite(metadata),
+            None => Interest::always(),
+        }
+    }
+
+    #[inline]
+    fn enabled(&self, metadata: &Metadata<'_>, ctx: Context<'_, S>) -> bool {
+        match self {
+            Some(ref inner) => inner.enabled(metadata, ctx),
+            None => true,
+        }
+    }
+
+    #[inline]
+    fn max_level_hint(&self) -> Option<LevelFilter> {
+        match self {
+            Some(ref inner) => inner.max_level_hint(),
+            None => None,
+        }
+    }
+
+    #[inline]
+    fn on_record(&self, _span: &span::Id, _values: &span::Record<'_>, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_record(_span, _values, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_follows_from(&self, _span: &span::Id, _follows: &span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_follows_from(_span, _follows, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_event(&self, _event: &Event<'_>, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_event(_event, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_enter(&self, _id: &span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_enter(_id, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_exit(&self, _id: &span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_exit(_id, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_close(&self, _id: span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_close(_id, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_id_change(&self, _old: &span::Id, _new: &span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_id_change(_old, _new, _ctx)
+        }
+    }
```

Comment:
We probably also want an implementation of `downcast_raw`:
https://github.com/tokio-rs/tracing/blob/247b6cc0a2e2427adbc1bd9df513545b6ef694ff/tracing-subscriber/src/layer.rs#L509-L516

I think that for `Option`s, it would look like this:
```suggestion
    }
    
    #[doc(hidden)]
    unsafe fn downcast_raw(&self, id: TypeId) -> Option<*const ()> {
       if id == TypeId::of::<Self>() {
            Some(self as *const _ as *const ())
        } else {
            self.and_then(|inner| inner.downcast_raw(id))
        }
    }
```